subprojects/plugins/src/main/java/org/gradle/api/plugins/BasePlugin.java

```diff
@@ -92,6 +92,8 @@ private void configureConfigurations(final Project project) {
         defaultConfiguration.deprecateForResolution(JavaPlugin.COMPILE_CLASSPATH_CONFIGURATION_NAME, JavaPlugin.RUNTIME_CLASSPATH_CONFIGURATION_NAME);
         archivesConfiguration.deprecateForDeclaration(JavaPlugin.IMPLEMENTATION_CONFIGURATION_NAME, JavaPlugin.API_CONFIGURATION_NAME);
         defaultConfiguration.deprecateForDeclaration(JavaPlugin.IMPLEMENTATION_CONFIGURATION_NAME, JavaPlugin.API_CONFIGURATION_NAME);
+        archivesConfiguration.deprecateForConsumption(JavaPlugin.RUNTIME_ELEMENTS_CONFIGURATION_NAME, JavaPlugin.API_ELEMENTS_CONFIGURATION_NAME);
```

Comment:
Should we postpone this to 7.1?

Reply:
I wanted to see what break if we do this. Looks there are some tests to update and to probably other things to double check. Lets do that in 7.1. I added a comment here: https://github.com/gradle/gradle/issues/15639#issuecomment-763606026